File: Models/Medicine.cs
using System.ComponentModel.DataAnnotations;
using System;
namespace SampleWebApp
{
    public class Medicine
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string FullName { get; set; }
        [Required]
        public string Notes { get; set; }
        [Required]
        public DateTime ExpDate { get; set; }
        [Required]
        public int Quantity { get; set; }
    }
}

File: ClientApp/src/app/medical/Medicine.ts
export interface Medicine {
  isSelected: boolean;
  id: number;
  fullName: string;
  notes: string;
  expDate: string;
  quantity: number;
  isEdit: boolean;
}

export const MedicineColumns = [
  {
    key: 'isSelected',
    type: 'isSelected',
    label: '',
  },
  {
    key: 'fullName',
    type: 'text',
    label: 'Full Name',
    required: true,
  },
  {
    key: 'notes',
    type: 'text',
    label: 'Notes',
    required: true,
  },
  {
    key: 'quantity',
    type: 'email',
    label: 'Quantity',
    required: true,
  },
  {
    key: 'expDate',
    type: 'date',
    label: 'Exp Date',
    required: true,
  },
  {
    key: 'isEdit',
    type: 'isEdit',
    label: '',
  },
];
